### PR TITLE
Update AVA

### LINKED ISSUE #251: Update tests to use AVA 3
Config can be copied from https://github.com/sindresorhus/refined-github/blob/main/package.json

The `esm` dependency should be dropped and I think the package type should be changed to `module`, either via env or in `package.json` (the latter would probably require extra changes though)

More details might be found in https://github.com/avajs/ava/issues/2293

### PR DESCRIPTION
**Pull request by fregante**
Fixes #251 

Follows and is based on #253

### CODE REVIEW

File: test/api-test.js [hunk 1]
```diff
@@ -190,7 +190,9 @@ test.serial('#makeApiRequest returns rejected promise for 4xx status codes', asy
 		statusText: 'Not found'
 	});
 
-	await t.throwsAsync(() => service.makeApiRequest('notifications'), 'client error');
+	await t.throwsAsync(() => service.makeApiRequest('notifications'), {
```
Comment (sindresorhus) [inline test/api-test.js]: Remove the `() => ` part.

Comment (fregante) [reply test/api-test.js]: How can `t.throwsAsync` catch and compare the error if the call throws before its result is passed to `t.throwsAsync`? 🤔

Comment (sindresorhus) [reply test/api-test.js]: It doesn't throw. `makeApiRequest` returns a promise. You pass a promise to `throwsAsync` and it checks whether it rejects or not.

Comment (fregante) [reply test/api-test.js]: Promises, how do they work

**Comment by fregante** [comment]
@notlmn It turns out most of the changes required for ESM support were done by XO in #253 

It pays to update the dependencies!

**Comment by notlmn** [comment]
@fregante thank you for this while I am MIA out of OSS!

**Pull request merged**